### DISCUSSION TITLE
Hiding IP address in public python program

### DISCUSSION THREAD

**Discussion by SketchedDoughnut** [community/community#123688] [Programming Help]
### Body

I want to preface this by saying that my Python knowledge is not the best, and I have _no_ experience with networking. 

So, I want to develop a program in Python that connects to a public IP address of a server (I am using an odroid c2)  hosting its own WiFi for a multiplayer game idea I had. This code is all public in a repository, and is not encrypted. 

I have two issues:
- I am struggling to find out how to hide that IP address, so that people can't see it, while also being able to use that IP to connect to the server. Or, alternatively, find other ways that are secure and don't expose the location of the server.
- I am not sure how to connect to devices outside of my own network in Python, as I have only previously watched a sockets tutorial which only let me connect to devices within my own network.

### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by kazmatw** [comment]
Hi @SketchedDoughnut ✋
# Here's how to address the two main concerns:

## Hiding the IP Address in Public Code:

- Use Environment Variables: Store the IP address in an environment variable. This way, you can refer to this variable in your Python code without revealing the actual IP address. On the server and any other system that needs to run the code, set this environment variable with the actual IP address.
- Configuration Files: Similar to environment variables, you can use a configuration file that is not included in the public repository. Read the IP address from this config file at runtime.
- Use a DNS Name: Instead of hardcoding the IP address, use a DNS name that resolves to the IP address of the server. This DNS can be updated without changing the code if the server IP changes.
## Connecting to External Devices Securely:

- VPN: Use a Virtual Private Network (VPN) to create a secure network tunnel. Both your server and the client machines can connect to the VPN. This hides the real IP addresses of all connected devices from the outside internet.
- Reverse Proxy: Set up a reverse proxy (like Nginx or Apache) that can act as an intermediary for requests from clients to the server. This hides the server's actual IP address and can also provide additional layers of security like SSL/TLS.
- Firewall and Security Groups: Ensure that the server is protected by firewalls and security settings that restrict which IP addresses and ports are open for connections.
